signer/tests/integration/request_decider.rs

```diff
@@ -398,3 +405,129 @@ async fn persist_received_deposit_decision_fetches_missing_deposit_requests() {
 
     testing::storage::drop_db(db).await;
 }
+
+/// Test `RequestDeciderEventLoop` behaviour in case of blocklist client
+/// failures. It should try to contact the blocklist client twice per bitcoin
+/// block, and in case of errors it should try again at the next block without
+/// voting.
+#[test_case::test_case(0, 0; "0 failures")]
+#[test_case::test_case(1, 0; "1 failure")]
+#[test_case::test_case(2, 1; "2 failures")]
```

Comment:
This is probably a bigger change, but can we add a test for the (2, 0) case where we end with zero rows in the database?

Reply:
We can add it, but IIUC this is already tested in the `#[test_case::test_case(2, 1; "2 failures")]` case, as for each failed iteration we check that we still have no votes (https://github.com/stacks-network/sbtc/pull/1270/files#diff-a8d19ef5983a5c5c6882f674d7b18dfd9326f0d7f25babfd863a986d000847a0R519). Are you referring to a different check?

Reply:
Ahh right! Yeah never mind.